homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,447 @@
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_AUX_HEAT,
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_HUMIDITY,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    ATTR_TARGET_TEMP_STEP,
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_AUX_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    STATE_OFF,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_AIRCLEANER_MODE,
+    ATTR_DEHUMIDIFY_SETPOINT,
+    ATTR_DEHUMIDIFY_SUPPORTED,
+    ATTR_HOLD_MODES,
+    ATTR_HUMIDIFY_SETPOINT,
+    ATTR_HUMIDIFY_SUPPORTED,
+    ATTR_SETPOINT_STATUS,
+    ATTR_ZONE_STATUS,
+    ATTRIBUTION,
+    DATA_NEXIA,
+    DOMAIN,
+    MANUFACTURER,
+    NEXIA_DEVICE,
+    UPDATE_COORDINATOR,
+)
+from .entity import NexiaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_SET_AIRCLEANER_MODE = "set_aircleaner_mode"
+SERVICE_SET_HUMIDIFY_SETPOINT = "set_humidify_setpoint"
+
+SET_FAN_MIN_ON_TIME_SCHEMA = vol.Schema(
+    {
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_AIRCLEANER_MODE): cv.string,
+    }
+)
+
+SET_HUMIDITY_SCHEMA = vol.Schema(
+    {
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_HUMIDITY): vol.All(
+            vol.Coerce(int), vol.Range(min=35, max=65)
+        ),
+    }
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up climate for a Nexia device."""
+
+    nexia_data = hass.data[DOMAIN][config_entry.entry_id][DATA_NEXIA]
+    nexia_home = nexia_data[NEXIA_DEVICE]
+    coordinator = nexia_data[UPDATE_COORDINATOR]
+
+    entities = []
+    for thermostat_id in nexia_home.get_thermostat_ids():
+        thermostat = nexia_home.get_thermostat_by_id(thermostat_id)
+        for zone_id in thermostat.get_zone_ids():
+            zone = thermostat.get_zone_by_id(zone_id)
+            entities.append(NexiaZone(coordinator, zone))
+
+    async_add_entities(entities, True)
+
+
+class NexiaZone(NexiaEntity, ClimateDevice):
+    """Provides Nexia Climate support."""
+
+    def __init__(self, coordinator, device):
+        """Initialize the thermostat."""
+        super().__init__(coordinator)
+        self.thermostat = device.thermostat
+        self._device = device
+        self._coordinator = coordinator
+        self._unique_id = f"{self._device.zone_id}_zone"
```

Comment:
I don't know for sure if there is an id collision between thermostat id and zone id as nexia seems to be just incrementing a global counter for newly added device.  There isn't clarity if the counter is per device type or a global so I added "_zone" so it would never collide with the thermostat id.

Reply:
But there's no entity added for the thermostat just for the zones. And the entity registry is aware of both the integration domain and the platform domain so there's no risk of collision with other platforms.

If there is more than one thermostat we want to add an id that is unique per thermostat, right, besides the zone id?

Reply:
The zone id is unique across the nexia platform and since we don't have an id for the thermostat, removing zone is safe.

Good call

Reply:
fixed